### ISSUE TITLE
Don't print an error on unused params in ES5 setters

### ISSUE THREAD

**Issue by mzgol** [eslint/eslint#2133]
The following code:
```js
(function () {
	"use strict";

	var o = {
		set b(a) {
		}
	};

	o.b = 2;
})();
```
triggers the error:
```
5:8 - a is defined but never used
```

In ES5 setters parameter is required, though and you can imagine a scenario where you want to use a setter without using the parameter.

This is similar to `try {...} catch (e) {}` where you need the parameter as well even if you don't need it. ESLint doesn't error on unused `catch` parameter.

The example was run on http://eslint.org/demo/.

**Comment by BenoitZugmeyer** [comment]
Just passing by:
> though and you can imagine a scenario where you want to use a setter without using the parameter.

I can't imagine such scenario and I'm curious, what is your use case?

**Comment by mzgol** [comment]
> I can't imagine such scenario and I'm curious, what is your use case?

One example is if you have to work in non-strict code and you still want to have a particular setter to throw:
```js
var o = {
    get x() {return 2;}
    set x(v) {throw new Error('o.x cannot be modified!');}
};
console.log(o.x);
o.x = 3;
```

**Comment by BenoitZugmeyer** [comment]
Nice, thank you!

**Comment by nzakas** [comment]
Yeah, I think the analogy of `catch` is spot-on. You literally have no choice but to provide the argument even if you arent going to use it. We should just skip checking it.

**Comment by nzakas** [comment]
Working on this.
